cpp/src/aztec3/circuits/rollup/merge/.test.cpp

```diff
@@ -9,41 +9,50 @@ using aztec3::circuits::rollup::native_merge_rollup::BaseOrMergeRollupPublicInpu
 using aztec3::circuits::rollup::native_merge_rollup::merge_rollup_circuit;
 using aztec3::circuits::rollup::native_merge_rollup::MergeRollupInputs;
 using aztec3::circuits::rollup::native_merge_rollup::NT;
+using DummyComposer = aztec3::utils::DummyComposer;
 
 } // namespace
 namespace aztec3::circuits::rollup::merge::native_merge_rollup_circuit {
 
 class merge_rollup_tests : public ::testing::Test {};
 
-#ifndef __wasm__ // TODO: temporary hack. WASM doesn't support EXPECT_DEATH or try/catch or EXPECT_THROW so no way to
-                 // test these in wasm.
 TEST_F(merge_rollup_tests, test_different_rollup_type_fails)
 {
+    DummyComposer composer = DummyComposer();
     auto mergeInput = dummy_merge_rollup_inputs_with_vk_proof();
     mergeInput.previous_rollup_data[0].base_or_merge_rollup_public_inputs.rollup_type = 0;
     mergeInput.previous_rollup_data[1].base_or_merge_rollup_public_inputs.rollup_type = 1;
-    EXPECT_DEATH(merge_rollup_circuit(mergeInput), ".*assert_both_input_proofs_of_same_rollup_type.*");
+    merge_rollup_circuit(composer, mergeInput);
+    ASSERT_TRUE(composer.has_failed());
+    ASSERT_EQ(composer.get_first_failure(), "input proofs are of different rollup types");
```

Comment:
It actually looks nice and tidy in the tests. (It's just a shame our actual circuit files are now riddled with the word `composer` everywhere!!!) :)

Reply:
Hmm we could have some new class and construct that has a composer member and all of these functions operate on an instance of that class.... But maybe that is overkill? Unless we think it would be cleaner in other ways too :thinking:

Reply:
Please could you elaborate on this idea?

Reply:
`ASSERT_TRUE` and `ASSERT_EQ` are gtest functions. I think it's nice to keep the gtest functions visible in the test files, rather than trying to hide them away. We want to be able to see the innards of what's being tested, to avoid missing a test.

Reply:
I meant that we have something like a NativeMergeRollup class that has a composer as a member and then each function in the class can use the composer without it being passed around via function args.

This is overkill though unless we see some other benefits to creating such a class.